### DISCUSSION TITLE
How to change the color of the keyword

### DISCUSSION THREAD

**Discussion by jiashengguo** [eclipse-langium/langium#1224] [Q&A] [answered]
By default, the color of the keywords in the **'Dark Modern'** theme  is purple:
<img width="236" alt="image" src="https://github.com/eclipse-langium/langium/assets/16688722/b35491c6-febd-496f-9f32-45d3a5c5fb9a">

I would like it to be blue which is the same as for TS:
![image](https://github.com/eclipse-langium/langium/assets/16688722/725efe56-47b4-492d-b0e8-e9105040d547)

How I can achieve so?  

BTW, I'm curious why the colors are the same for some color themes like '**Dark(Visual Studio)'**(all blue)

**Comment by msujew** [comment, marked as answer]
@jiashengguo These are just done using different scopes for the textMate tokenizer. You can inspect the scope of a token in vscode by executing the `Developer: Inspect Editor Tokens and Scopes` command.

To change this you need to modify the generated textMate grammar to use a different scope for your keywords. Since Langium doesn't really know which keyword belongs to which scope (see also the discussion over at https://github.com/eclipse-langium/langium/issues/818), it will just generate all keywords into the default keyword scope (`keyword.control.<id>`).

To use the blue colored keywords in your example, you need to switch from the `keyword.control.<id>` to the `keyword.storage.<id>` scope.

> BTW, I'm curious why the colors are the same for some color themes like 'Dark(Visual Studio)'(all blue)

In vscode, it's completely up to the theme designer to change the color certain scopes use. The language server/textMate grammars have no influence over the actual coloring, they just give the editor some syntactic/semantic token data. The actual coloring is purely depended on the theme.

**Reply by jiashengguo** [reply]
@msujew  Thank you so much! Not only did you answer my question, but you also taught me more!

It worked after I directly changed it in the tmLanguage.json file. However, it will be reverted back if I regenerate it with `langium generate`. Is there a way to make this change permanent?

**Reply by msujew** [reply]
You're welcome :)

> Is there a way to make this change permanent?

Not as of now. Our recommendation is generally to use the generated textmate/syntax highlighting files as a starting point for prototyping and create a hand-written syntax highlighting file for production usage. We do the same for our [own Langium DSL](https://github.com/eclipse-langium/langium/blob/main/packages/langium-vscode/data/langium.tmLanguage.json), since it requires more advanced stuff such as regex highlighting.

**Reply by jiashengguo** [reply]
Got it.  One more thing learned from you.  Thanks!
